packages/lib-core/src/api/common-types.ts

```diff
@@ -0,0 +1,40 @@
+import type { AnyObject } from '@monorepo/common';
+
+// Type for extension hook
+export type ExtensionHook<TType, TOptions extends AnyObject = AnyObject> = (
+  options: TOptions,
+) => ExtensionHookResult<TType>;
+
+// Type for extension hook result that returns [data, resolved, error]
+export type ExtensionHookResult<TType> = [TType, boolean, unknown];
+
+export type ExtensionK8sKindVersionModel = {
+  group?: string;
+  version: string;
+  kind: string;
+};
+
+export type TypeAndTitle = {
+  /** Type ID for the item type. */
+  type: string;
+  /** Title for the item */
+  title: string | React.ReactNode;
+};
+
+export type ExtensionK8sModel = {
+  group: string;
+  version: string;
+  kind: string;
+};
+
+export type ExtensionK8sGroupKindModel = {
+  group: string;
+  version?: string;
+  kind: string;
+};
+
+export type ExtensionK8sGroupModel = {
+  group: string;
+  version?: string;
+  kind?: string;
+};
```

Comment:
@andrewballantyne AFAIK this was discussed with @spadgett and the consensus was to have a single way to identify k8s resources - check out `K8sResourceIdentifier` type in #30 - group is optional but version and kind are mandatory:

```ts
type K8sResourceIdentifier = {
  apiGroup?: string;
  apiVersion: string;
  kind: string;
};
```

Notice that k8s APIs use `apiGroup` instead of `group` and `apiVersion` instead of `version` - I wouldn't change that in `K8sResourceIdentifier` type, as they reflect the k8s API response data.

TL;DR - I'd suggest introducing a single type via e.g. `packages/lib-core/src/types/extension-k8s.ts` which is conceptually equivalent to above.

```ts
type ExtensionK8sResourceIdentifier = {
  group?: string;
  version: string;
  kind: string;
};
```

Reply:
It's technically possible to reuse `K8sResourceIdentifier` from `lib-utils` and rename the group/version property names (like [this](https://stackoverflow.com/a/52702528)) but that would introduce `lib-core` => `lib-utils` (or the opposite) dependency, which we generally want to avoid.

Core k8s types should be kept in `lib-utils` since this package deals primarily with k8s API calls, among other things.